src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/AttributeValueResolver.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Attribute\ValueResolver;
+use Symfony\Component\HttpKernel\Controller\ValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+
+/**
+ * Delegates value resolving of arguments bearing the {@see ValueResolver} attribute to the specified resolver.
+ *
+ * @author Mathieu Lechat <mathieu.lechat@les-tilleuls.com>
+ */
+final class AttributeValueResolver implements ValueResolverInterface
+{
+    public function __construct(private readonly ContainerInterface $resolvers)
+    {
+    }
+
+    public function resolve(Request $request, ArgumentMetadata $argument): iterable
+    {
+        $attributes = $argument->getAttributesOfType(ValueResolver::class);
+
+        if (!$attributes) {
+            return [];
+        }
+
+        $resolverName = $attributes[0]->name;
+        $resolver = $this->resolvers->get($resolverName);
+        if (!$resolver instanceof ValueResolverInterface) {
+            throw new \DomainException(sprintf('"%s" resolver must implement "%s", but got "%s".', $resolverName, ValueResolverInterface::class, get_debug_type($resolver)));
+        }
+
+        return $resolver->resolve($request, $argument);
```

Comment:
If the resolver returns an empty array (meaning it did not support the argument), we should probably throw an exception instead of letting next value resolvers run (adding a ValueResolver attribute on the argument means we expect that resolver to be used)

Reply:
Given the resolver could return an empty `Traversable`, do we agree we have to transform it to an array in that case?

Reply:
We could, but then we need to return the converted array, not the original iterator (in case it is a non-rewindable one due to using a generator)

Reply:
Indeed, took care of it.